app/services/backup_code_generator.rb

```diff
@@ -52,19 +60,24 @@ def code_usable?(code)
     code && code.used_at.nil?
   end
 
-  def save_code(code)
-    @user.backup_code_configurations.create!(code: code)
-  end
-
-  def normalize(plaintext_code)
-    plaintext_code.gsub(/\W/, '').delete('-').downcase.strip
+  def save_code(code:, salt:)
+    @user.backup_code_configurations.create!(
+      code_salt: salt,
+      code_cost: cost,
+      skip_legacy_encryption: skip_legacy_encryption?,
+      code: code,
+    )
   end
 
   def backup_code
-    normalize(SecureRandom.hex(@split * @length / 2))
+    RandomPhrase.new(num_words: num_words, separator: nil).to_s
```

Comment:
Updating to use RandomPhrase, which has the advantage of filtering out profanity now that https://github.com/18F/identity-idp/pull/4888 has been merged